mesonbuild/utils/universal.py

```diff
@@ -1362,10 +1362,17 @@ def do_conf_file(src: str, dst: str, confdata: 'ConfigurationData',
 
 '''
 
-def _dump_c_header(ofile: T.TextIO, cdata: ConfigurationData, output_format: Literal['c', 'nasm']) -> None:
+def _dump_c_header(ofile: T.TextIO,
+                   cdata: ConfigurationData,
+                   output_format: Literal['c', 'nasm'],
+                   macro_name: str,
+                   guard_type: Literal['pragma', 'macro']) -> None:
```

Comment:
This API can be simplified by only accepting a `macro_name: T.Optional[str]` and using None as the sentinel for "use a pragma".